File: Misc/coin_flips_multi_lang/coin_flip.rb
# frozen_string_literal: true

heads_count = 0
tails_count = 0
max_iterations = 100_000

(0..max_iterations).each do
  rand(1..2) == 1 ? heads_count += 1 : tails_count += 1
end

puts('Heads #: ' + String(heads_count))
puts('Tails #: ' + String(tails_count))

File: Misc/coin_flips_multi_lang/coin_flip.cpp
#include <iostream>

int main() {
    int heads_count = 0;
    int tails_count = 0;
    int coin_flips = 100'000'000;
    
    for (int i = 0; i < coin_flips; i++) {
        ((rand() % 2 + 1) == 1) ? heads_count++ : tails_count++;
    }
    
    std::cout << "Tails #: " << tails_count;
    std::cout << "\n";
    std::cout << "Heads #: " << heads_count;
    return 0;
}
